packages/cactus-test-tooling/src/main/typescript/daml/daml-test-ledger.ts

```diff
@@ -0,0 +1,328 @@
+    return `http://${ipAddress}:${hostPort}`;
+  }
+
+  public async getDamlApiHost(): Promise<string> {
+    const ipAddress = "127.0.0.1";
+    // const port = await Containers.getPublicPort(rpcApiWsPort, containerInfo);
+    const port = "7575";
+    return `http://${ipAddress}:${port}`;
+  }
+
+  public async getFileContents(filePath: string): Promise<string> {
+    const response = await this.getContainer().getArchive({
+      path: filePath,
+    });
+    const extract: tar.Extract = tar.extract({ autoDestroy: true });
+
+    return new Promise((resolve, reject) => {
+      let fileContents = "";
+      extract.on("entry", async (header: unknown, stream, next) => {
+        stream.on("error", (err: Error) => {
+          reject(err);
+        });
+        const chunks: string[] = await Streams.aggregate<string>(stream);
+        fileContents += chunks.join("");
+        stream.resume();
+        next();
+      });
+
+      extract.on("finish", () => {
+        resolve(fileContents);
+      });
+
+      response.pipe(extract);
+    });
+  }
+
+  public async start(omitPull = false): Promise<Container> {
+    const imageFqn = this.getContainerImageName();
+
+    if (this.container) {
+      await this.container.stop();
+      await this.container.remove();
+    }
+    const docker = new Docker();
+
+    if (!omitPull) {
+      this.log.debug(`Pulling container image ${imageFqn} ...`);
+      await this.pullContainerImage(imageFqn);
+      this.log.debug(`Pulled ${imageFqn} OK. Starting container...`);
+    }
+
+    return new Promise<Container>((resolve, reject) => {
+      const eventEmitter: EventEmitter = docker.run(
+        imageFqn,
+        [],
+        [],
+        {
+          NetworkMode: "host",
+          ExposedPorts: {
+            "7575/tcp": {}, // DAML API - HTTP
+          },
+          // TODO: this can be removed once the new docker image is published and
+          // specified as the default one to be used by the tests.
+          HostConfig: {
+            PublishAllPorts: true,
+          },
+        },
+        {},
+        (err: unknown) => {
+          if (err) {
+            reject(err);
+          }
+        },
+      );
+
+      eventEmitter.once("start", async (container: Container) => {
+        this.log.debug(`Started container OK. Waiting for healthcheck...`);
+        this.container = container;
+        this.containerId = container.id;
+
+        if (this.emitContainerLogs) {
+          const fnTag = `[${this.getContainerImageName()}]`;
+          await Containers.streamLogs({
+            container: this.getContainer(),
+            tag: fnTag,
+            log: this.log,
+          });
+        }
+
+        try {
+          await this.waitForHealthCheck();
+          this.log.debug(`Healthcheck passing OK.`);
+          resolve(container);
+        } catch (ex) {
+          reject(ex);
+        }
+      });
+    });
+  }
+
+  public async waitForHealthCheck(timeoutMs = 360000): Promise<void> {
```

Comment:
5 minutes looks like a very big timeout; perhaps reduce it accordin g to other test ledgers?

Reply:
Hello @RafaelAPB ,
At the moment I set this to 2 minutes. However I will still do some runs if this can still be reduced. Thanks!

Cheers!
Rayn